flee/flee.py

```diff
@@ -93,10 +88,10 @@ def evolve(self, time: int, ForceTownMove: bool = False) -> None:
                     chosenRoute = self.selectRouteRuleset2(time=time)
 
                 # if there is a viable route to a different location.
-                if chosenRoute >= 0:
+                if chosenRoute:
                     # update location to link endpoint
                     self.location.DecrementNumAgents()
-                    self.location = self.location.links[chosenRoute]
+                    self.location = chosenRoute
```

Comment:
Instead of returning an index to then look up the location we returned the location. This reduced one expensive call.